app/authentication/auth.py

```diff
@@ -54,21 +60,29 @@ def restrict_ip_sms():
             ip = ip_list[len(ip_list) - 3]
         current_app.logger.info("Inbound sms ip route list {}"
                                 .format(ip_route))
+    p0 = ip.split('.')
 
-    # Temporary custom header for route security - to experiment if the header passes through
-    if request.headers.get("X-Custom-forwarder"):
-        current_app.logger.info("X-Custom-forwarder {}".format(request.headers.get("X-Custom-forwarder")))
+    # IP whitelist
+    allowed_ips = current_app.config.get('SMS_INBOUND_WHITELIST')
+    allowed = False
+
+    for allowed_ip in allowed_ips:
+        p1 = allowed_ip.split('.')
+        if p0[0] == p1[0] and p0[1] == p1[1] and p0[2] == p1[2]:
+            allowed = True
```

Comment:
i'd rather we use the [ipaddress library](https://docs.python.org/3/library/ipaddress.html) for this.

And then, we should update the `SMS_INBOUND_WHITELIST` variable to contain address masks (eg `192.168.0.0/24`) rather than whole IPs.